paddle/fluid/framework/ir/memory_optimize_pass/recurrent_op_eager_deletion_pass.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <unordered_map>
+
+#include "paddle/fluid/framework/details/computation_op_handle.h"
+#include "paddle/fluid/framework/details/multi_devices_helper.h"
+#include "paddle/fluid/framework/ir/graph_helper.h"
+#include "paddle/fluid/operators/controlflow/op_variant.h"
+#include "paddle/fluid/operators/controlflow/recurrent_op_helper.h"
+
+namespace paddle {
+namespace framework {
+namespace details {
+
+static constexpr char kStepBlock[] = "sub_block";
+static constexpr char kStepScopes[] = "step_scopes";
+static constexpr char kInputs[] = "inputs";
+static constexpr char kInputsGrad[] = "inputs@GRAD";
+static constexpr char kOutputs[] = "outputs";
+static constexpr char kSkipEagerDeletionVars[] = "skip_eager_deletion_vars";
+
+using paddle::operators::OpVariant;
+using paddle::operators::OpVariantSet;
+using paddle::operators::OpAndGradOpPair;
+
+// Pass class set skip eager deletion vars for recurrent ops
+class RecurrentOpEagerDeletionPass : public ir::Pass {
```

Comment:
`RecurrentOpEagerDeletionPass` can be placed inside `recurrent_op_eager_deletion_pass.cc`. Therefore, this header file is unnecessary.

Reply:
Can I keep it? I prefer every .cc file should have an associated .h file in general, except for some special cases.